examples/build_file_generation/WORKSPACE

```diff
@@ -28,45 +36,97 @@ http_archive(
 
+# go_rules_dependencies is a function that registers external dependencies
+# needed by the Go rules.
+# https://github.com/bazelbuild/rules_go/blob/master/go/dependencies.rst#go_rules_dependencies
 go_register_toolchains(version = "1.18.3")
 
+# The following call configured the gazelle dependencies, Go environment and Go SDK.
 gazelle_dependencies()
 
-######################################################################
 # Remaining setup is for rules_python
 
+# You do not want to use the following command when you are using a WORKSPACE file
+# that is outside of rules_python repository.
+# This command allows targets from a local directory to be bound.
+# Which allows bazel to use targets defined in base rules_python directory.
+# If you are using this example outside of the rules_python git repo,
+# use the http_archive command that is commented out below.
+# https://bazel.build/reference/be/workspace#local_repository
 local_repository(
     name = "rules_python",
     path = "../..",
 )
 
+# When not using this example in the rules_python git repo you would load the python
+# ruleset using the following StarLark.
+# See https://github.com/bazelbuild/rules_python#getting-started for the latest
+# ruleset version.
+#
+# The following StarLark would replace the `local_repository` rule mentioned above.
+#
+# rules_python_version = "740825b7f74930c62f44af95c9a4c1bd428d2c53" # Latest @ 2021-06-23
+#
+# http_archive(
+#     name = "rules_python",
+#     sha256 = "3474c5815da4cb003ff22811a36a11894927eda1c2e64bf2dac63e914bfdf30f",
+#     strip_prefix = "rules_python-{}".format(rules_python_version),
+#     url = "https://github.com/bazelbuild/rules_python/archive/{}.zip".format(rules_python_version),
+# )
+
+# Next we load the toolchain from rules_python.
 load("@rules_python//python:repositories.bzl", "python_register_toolchains")
 
+# We now register a hermetic Python interpreter rather than relying on a system-installed interpreter.
+# This toolchain will allow bazel to download a specific python version, and use that version
+# for compilation.
 python_register_toolchains(
     name = "python39",
     python_version = "3.9",
 )
 
+# load the interpreter and pip_parse rules.
 load("@python39//:defs.bzl", "interpreter")
 load("@rules_python//python:pip.bzl", "pip_parse")
 
+# This macro wraps the `pip_repository` rule that invokes `pip`, with `incremental` set.
+# Accepts a locked/compiled requirements file and installs the dependencies listed within.
+# Those dependencies become available in a generated `requirements.bzl` file.
+# You can instead check this `requirements.bzl` file into your repo.
 pip_parse(
     name = "pip",
+    # (Optional) You can provide a python_interpreter (path) or a python_interpreter_target (a Bazel target, that
+    # acts as an executable). The latter can be anything that could be used as Python interpreter. E.g.:
+    # 1. Python interpreter that you compile in the build file (as above in @python_interpreter).
```

Comment:
I think this comment is incorrect. `as above in @python_interpreter` refers to nothing.